### PR TITLE
Add mtu and accessConfig flags to subnet create commands for terraform

### PR DESCRIPTION
**Pull request by ismirlia**
This pull request makes the necessary updates to the terraform provider to support these two new options in subnet create.

Unsupported endpoints are safeguarded in the SDK, and there are a few checks extra checks for the fields that are not supposed to be set for the endpoints that changed (subnet create and instance create).

```
=== RUN   TestAccIBMPINetworkGatewaybasic
--- PASS: TestAccIBMPINetworkGatewaybasic (70.68s)
PASS


=== RUN   TestAccIBMPINetworkGatewaybasicSatellite
--- PASS: TestAccIBMPINetworkGatewaybasicSatellite (48.31s)
PASS
```

### CODE REVIEW

**Review by powervs-ibm — APPROVED** [review]
Looks good

File: ibm/service/power/resource_ibm_pi_network.go [hunk 1]
```diff
@@ -80,6 +80,18 @@ func ResourceIBMPINetwork() *schema.Resource {
 				Computed:    true,
 				Description: "PI network enable MTU Jumbo option",
```
Comment (michaelkad) [inline ibm/service/power/resource_ibm_pi_network.go:81]:
```suggestion
				ExactlyOneOf: []string{ helpers.PINetworkMtu, helpers.PINetworkJumbo},
				Description: "PI network enable MTU Jumbo option",
				
```

File: ibm/service/power/resource_ibm_pi_network.go [hunk 2]
```diff
@@ -80,6 +80,18 @@ func ResourceIBMPINetwork() *schema.Resource {
...
 			},
+			helpers.PINetworkMtu: {
+				Type:        schema.TypeInt,
+				Optional:    true,
+				Computed:    true,
+				Description: "PI Maximum Transmission Unit",
+			},
```
Comment (michaelkad) [inline ibm/service/power/resource_ibm_pi_network.go:88]:
```suggestion
			helpers.PINetworkMtu: {
				Type:        schema.TypeInt,
				Optional:    true,
				Computed:    true,
				ExactlyOneOf: []string{ helpers.PINetworkMtu, helpers.PINetworkJumbo},
				Description: "PI Maximum Transmission Unit",
			},
```

**Review by michaelkad — COMMENTED** [review]
@ismirlia `mtu` will be used for both

File: website/docs/r/pi_network.html.markdown [hunk 3]
```diff
@@ -69,7 +69,9 @@ Review the argument references that you can specify for your resource.
   The `pi_ipaddress_range` block supports:
   - `pi_ending_ip_address` - (Required, String) The ending ip address.
   - `pi_starting_ip_address` - (Required, String) The staring ip address. **Note** if the `pi_gateway` or `pi_ipaddress_range` is not provided, it will calculate the value based on CIDR respectively.
-- `pi_network_jumbo` - (Optional, Bool) MTU Jumbo option of the network.
+- `pi_network_jumbo` - (Optional, Bool) MTU Jumbo option of the network (for multi-zone locations only).
+- `pi_network_mtu` - (Optional, Integer) Maximum Transmission Unit option of the network (for satellite locations only).
```
[the remark this answers is not in the document]
Comment (yussufsh) [reply website/docs/r/pi_network.html.markdown:73]: Is Network jumbo option really deprecated? I see this in the swagger def: `Enable MTU Jumbo Network (for multi-zone locations only)`. If you still think it needs deprecation then please add the `Deprecated` field in `pi_network_jumbo` definition in the resource as well.

[the remark this answers is not in the document]
Comment (michaelkad) [reply website/docs/r/pi_network.html.markdown:73]: Yes — it is being deprecated @yussufsh 
@ismirlia Could you please add deprecated to the `pi_network_jumbo` as well?

**Comment by michaelkad** [comment]
@hkantare @yussufsh  Could you help with reviewing this PR please?

File: ibm/service/power/resource_ibm_pi_network.go [hunk 4]
```diff
@@ -75,10 +75,25 @@ func ResourceIBMPINetwork() *schema.Resource {
...
+				Description:  "PI network enable MTU Jumbo option",
+			},
+			helpers.PINetworkMtu: {
+				Type:         schema.TypeInt,
+				Optional:     true,
+				Computed:     true,
+				ExactlyOneOf: []string{helpers.PINetworkMtu, helpers.PINetworkJumbo},
+				Description:  "PI Maximum Transmission Unit",
+			},
+			helpers.PINetworkAccessConfig: {
```
Comment (yussufsh) [inline ibm/service/power/resource_ibm_pi_network.go:92]: Please add validation for all the possible values. https://github.com/IBM-Cloud/power-go-client/blob/master/power/models/access_config.go#L39-L51
Example:
```
helpers.PINetworkType: {
    Type:         schema.TypeString,
    Required:     true,
    ValidateFunc: validate.ValidateAllowedStringValues([]string{"vlan", "pub-vlan"}),
    Description:  "PI network type",
},
```

File: ibm/service/power/resource_ibm_pi_network_test.go [hunk 5]
```diff
@@ -88,6 +88,43 @@ func TestAccIBMPINetworkGatewaybasic(t *testing.T) {
 		},
 	})
 }
+
```
Comment (yussufsh) [inline ibm/service/power/resource_ibm_pi_network_test.go:91]: It was ok to update the existing basic test case. Just a suggestion... will let you decide, and mark it as resolved if you are good.

**Review by yussufsh — CHANGES_REQUESTED** [review]
minor comments.

File: ibm/service/power/resource_ibm_pi_network_test.go (hunk 5 above)
Comment (ismirlia) [reply ibm/service/power/resource_ibm_pi_network_test.go:91]: Please mark it as a future enhancement. I am currently out and will be out until next Tuesday.

**Review by yussufsh — APPROVED** [review]
lgtm
@hkantare please review this one as well. Thanks.

**Comment by ismirlia** [comment]
@hkantare Please review this when I get a chance so I can merge it.

**Comment by ismirlia** [comment]
@hkantare Are there any issues with merging this pr?

**Review by hkantare — APPROVED** [review]

**Comment by ismirlia** [comment]
Thanks!

**Pull request merged**